src/odemis/acq/stitching/_registrar.py

```diff
@@ -14,18 +14,442 @@
 
 You should have received a copy of the GNU General Public License along with Odemis. If not, see http://www.gnu.org/licenses/.
 '''
-from __future__ import division
-
 
 # This is a series of classes which use different methods to compute the "best
 # location" of an image set based on their metadata and content. IOW, it does
 # "image registration".
 
-# TODO: define a generic API: can take a set of images, with some being explicitly
-# linked/locked in position, and allow to add new images on the set (to run it
-# in live mode).
 
-# TODO: simple version which returns the data as-is
+from __future__ import division
+from odemis.acq.drift import CalculateDrift
+import numpy
+import math
+from odemis import model
+
+BEST_MATCH = 0.9  # minimum match value indicating optimal shift value
+GOOD_MATCH = 0.2  # minimum match value indicating shift value can be used in the average position calculation
+EXTREME_SHIFT = 0.4  # minimum ratio of overlap size indicating an extreme shift value that should be aborted
+
+
+        
+class IdentityRegistrar(object):
+    """ Returns position as-is """
+    
+    def __init__(self):
+        self.tiles = []
+        self.num_dep_tiles = 0
+        
+    def addTile(self,tile,dependent_tiles=None):
+        """ 
+        Extends grid by one tile. 
+        tile (DataArray of shape YX): each image must have at least MD_POS and MD_PIXEL_SIZE metadata. 
+        They should all have the same dtype.
+        dependent_tiles (list of DataArray or None): each of the dependent tile, where their position 
+        can be considered fixed relative to the main tile. Their content and metadata are not used
+        for the computation of the final position.
+        """
+        self.tiles.append(tile)
+        if dependent_tiles != None:
+            self.num_dep_tiles = len(dependent_tiles)
+        
+    def getPositions(self):
+        """ 
+        returns:
+        tile_positions (list of tuples): the adjusted position in X/Y for each tile, in the order they were added
+        dep_tile_positions (list of tuples of tuples): the adjusted position for each dependent tile 
```

Comment:
The ```dep_tile_positions``` description needs to clarify the order of the dimensions. I'd go for actually the opposite of what is implemented: ```list of N tuples of tuples of 2 floats```. The list has as many elements the tile_positions, and each element contains the positions for all the dependent tiles passed with the corresponding tile.

Unfortunately, that means you also need to reorganise your code a bit.